### ISSUE TITLE
Can't build on Docker

### ISSUE THREAD

**Issue by raphaelsetin** [assimp/assimp#3906]
**Describe the bug**
Can't build on AmazonLinux:2018.03 nor on Ubuntu:14.04 (from the repo's Dockerfile). I will show below the error that I got from my custom AmazonLinux Dockerfile. If you want to try the repo's Dockerfile and see the other bug, please feel free to do so.

**To Reproduce**
1. Create the following Dockerfile:
<pre>
FROM amazonlinux:2018.03

# Arguments
ARG CMAKE_VERSION_MAJOR="3"
ARG CMAKE_VERSION_MINOR="19"
ARG CMAKE_VERSION_PATCH="6"
# Assimp version date: May 17, 2021, basically it's the current master
ARG ASSIMP_HASH="0c9947f8b9012faa43a77917f1035db2ffc9a0d2"

# Set the user
USER root

# Install deps required
RUN yum update -y && \
	yum groupinstall -y "Development Tools" && \
	yum install openssl-devel git wget -y

# Install latest version of cmake required
# because yum provides just 2.8 and minimum 3.12 required
RUN wget https://cmake.org/files/v${CMAKE_VERSION_MAJOR}.${CMAKE_VERSION_MINOR}/cmake-${CMAKE_VERSION_MAJOR}.${CMAKE_VERSION_MINOR}.${CMAKE_VERSION_PATCH}.tar.gz && \
	tar -xvzf cmake-${CMAKE_VERSION_MAJOR}.${CMAKE_VERSION_MINOR}.${CMAKE_VERSION_PATCH}.tar.gz && \
	cd cmake-${CMAKE_VERSION_MAJOR}.${CMAKE_VERSION_MINOR}.${CMAKE_VERSION_PATCH} && \
	./bootstrap && \
	make && \
	make install

# Clone assimp from specific commit hash (since it's unstable) and build it
RUN git clone https://github.com/assimp/assimp.git && \
	cd assimp && \
	git reset --hard $ASSIMP_HASH && \
	cmake CMakeLists.txt -DBUILD_SHARED_LIBS=OFF -DASSIMP_BUILD_TESTS=OFF && \
	make -j4
</pre>
2. Run `docker build -t "assimp-test" .`
3. See the following error:
<pre>
#6 2.603 cmake-3.19.6/Tests/FindGTK2/giomm/
#6 2.603 cmake-3.19.6/Tests/FindGTK2/giomm/CMakeLists.txt
#6 2.603 cmake-3.19.6/Tests/FindGTK2/giomm/mai
#6 2.603 [output clipped, log limit 100KiB/s reached]
#6 DONE 730.7s

#7 [4/4] RUN git clone https://github.com/assimp/assimp.git &&  cd assimp &&   git reset --hard 0c9947f8b9012faa43a77917f1035db2ffc9a0d2 &&     cmake CMakeLists.txt -DBUILD_SHARED_LIBS=OFF -DASSIMP_BUILD_TESTS=OFF &&        make -j4
#7 sha256:20b6d1a20510820980635b8fa393f14940aa43df100537b82501ffb1094d3a1e
#7 0.381 Cloning into 'assimp'...
#7 12.93 HEAD is now at 0c9947f8 Merge pull request #3900 from Saalvage/decltype
#7 12.98 -- The C compiler identification is GNU 7.2.1
#7 13.04 -- The CXX compiler identification is GNU 4.8.5
#7 13.05 -- Detecting C compiler ABI info
#7 13.10 -- Detecting C compiler ABI info - done
#7 13.11 -- Check for working C compiler: /usr/bin/cc - skipped
#7 13.11 -- Detecting C compile features
#7 13.11 -- Detecting C compile features - done
#7 13.12 -- Detecting CXX compiler ABI info
#7 13.18 -- Detecting CXX compiler ABI info - done
#7 13.19 -- Check for working CXX compiler: /usr/bin/c++ - skipped
#7 13.19 -- Detecting CXX compile features
#7 13.26 -- Detecting CXX compile features - done
#7 13.26 -- Shared libraries disabled
#7 13.28 -- Looking for ZLIB...
#7 13.28 -- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1")
#7 13.28 -- Checking for module 'zzip-zlib-config'
#7 13.28 --   No package 'zzip-zlib-config' found
#7 13.29 -- Found ZLIB: optimized;/usr/lib64/libz.so;debug;/usr/lib64/libz.so
#7 13.29 -- Checking for module 'minizip'
#7 13.29 --   No package 'minizip' found
#7 13.30 -- Enabled importer formats: AMF 3DS AC ASE ASSBIN B3D BVH COLLADA DXF CSM HMP IRRMESH IRR LWO LWS M3D MD2 MD3 MD5 MDC MDL NFF NDO OFF OBJ OGRE OPENGEX PLY MS3D COB BLEND IFC XGL FBX Q3D Q3BSP RAW SIB SMD STL TERRAGEN 3D X X3D GLTF 3MF MMD
#7 13.30 -- Disabled importer formats:
#7 13.30 -- Enabled exporter formats: OBJ OPENGEX PLY 3DS ASSBIN ASSXML M3D COLLADA FBX STL X X3D GLTF 3MF PBRT ASSJSON STEP
#7 13.30 -- Disabled exporter formats:
#7 13.31 -- Configuring done
#7 13.37 -- Generating done
#7 13.37 -- Build files have been written to: /assimp
#7 13.72 Scanning dependencies of target assimp
#7 13.76 [  0%] Building CXX object code/CMakeFiles/assimp.dir/Common/Assimp.cpp.o
#7 13.76 [  0%] Building CXX object code/CMakeFiles/assimp.dir/CApi/CInterfaceIOWrapper.cpp.o
#7 13.76 [  1%] Building CXX object code/CMakeFiles/assimp.dir/Common/BaseImporter.cpp.o
#7 13.76 [  1%] Building CXX object code/CMakeFiles/assimp.dir/Common/BaseProcess.cpp.o
#7 14.15 [  2%] Building CXX object code/CMakeFiles/assimp.dir/Common/PostStepRegistry.cpp.o
#7 14.31 [  2%] Building CXX object code/CMakeFiles/assimp.dir/Common/ImporterRegistry.cpp.o
#7 14.39 In file included from /assimp/include/assimp/Logger.hpp:50:0,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /assimp/include/assimp/TinyFormatter.h: In instantiation of 'Assimp::Formatter::basic_formatter<T, CharTraits, Allocator>::basic_formatter(Assimp::Formatter::basic_formatter<T, CharTraits, Allocator>&&) [with T = char; CharTraits = std::char_traits<char>; Allocator = std::allocator<char>]':
#7 14.39 /assimp/include/assimp/Exceptional.h:77:82:   required from 'DeadlyImportError::DeadlyImportError(T&& ...) [with T = {const char (&)[18]}]'
#7 14.39 /assimp/code/Common/BaseImporter.cpp:341:52:   required from here
#7 14.39 /assimp/include/assimp/TinyFormatter.h:92:49: error: use of deleted function 'std::basic_ostringstream<char>::basic_ostringstream(const std::basic_ostringstream<char>&)'
#7 14.39          : underlying(std::move(other.underlying)) {
#7 14.39                                                  ^
#7 14.39 In file included from /assimp/include/assimp/TinyFormatter.h:56:0,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/sstream:387:11: note: 'std::basic_ostringstream<char>::basic_ostringstream(const std::basic_ostringstream<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 14.39      class basic_ostringstream : public basic_ostream<_CharT, _Traits>
#7 14.39            ^
#7 14.39 /usr/include/c++/4.8.5/sstream:387:11: error: use of deleted function 'std::basic_ostream<char>::basic_ostream(const std::basic_ostream<char>&)'
#7 14.39 In file included from /usr/include/c++/4.8.5/istream:39:0,
#7 14.39                  from /usr/include/c++/4.8.5/sstream:38,
#7 14.39                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/ostream:58:11: note: 'std::basic_ostream<char>::basic_ostream(const std::basic_ostream<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 14.39      class basic_ostream : virtual public basic_ios<_CharT, _Traits>
#7 14.39            ^
#7 14.39 /usr/include/c++/4.8.5/ostream:58:11: error: use of deleted function 'std::basic_ios<char>::basic_ios(const std::basic_ios<char>&)'
#7 14.39 In file included from /usr/include/c++/4.8.5/ios:44:0,
#7 14.39                  from /usr/include/c++/4.8.5/istream:38,
#7 14.39                  from /usr/include/c++/4.8.5/sstream:38,
#7 14.39                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/bits/basic_ios.h:66:11: note: 'std::basic_ios<char>::basic_ios(const std::basic_ios<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 14.39      class basic_ios : public ios_base
#7 14.39            ^
#7 14.39 In file included from /usr/include/c++/4.8.5/ios:42:0,
#7 14.39                  from /usr/include/c++/4.8.5/istream:38,
#7 14.39                  from /usr/include/c++/4.8.5/sstream:38,
#7 14.39                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/bits/ios_base.h:786:5: error: 'std::ios_base::ios_base(const std::ios_base&)' is private
#7 14.39      ios_base(const ios_base&);
#7 14.39      ^
#7 14.39 In file included from /usr/include/c++/4.8.5/ios:44:0,
#7 14.39                  from /usr/include/c++/4.8.5/istream:38,
#7 14.39                  from /usr/include/c++/4.8.5/sstream:38,
#7 14.39                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/bits/basic_ios.h:66:11: error: within this context
#7 14.39      class basic_ios : public ios_base
#7 14.39            ^
#7 14.39 In file included from /assimp/include/assimp/TinyFormatter.h:56:0,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/sstream:387:11: error: use of deleted function 'std::basic_ios<char>::basic_ios(const std::basic_ios<char>&)'
#7 14.39      class basic_ostringstream : public basic_ostream<_CharT, _Traits>
#7 14.39            ^
#7 14.39 /usr/include/c++/4.8.5/sstream:387:11: error: use of deleted function 'std::basic_stringbuf<char>::basic_stringbuf(const std::basic_stringbuf<char>&)'
#7 14.39 /usr/include/c++/4.8.5/sstream:64:11: note: 'std::basic_stringbuf<char>::basic_stringbuf(const std::basic_stringbuf<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 14.39      class basic_stringbuf : public basic_streambuf<_CharT, _Traits>
#7 14.39            ^
#7 14.39 In file included from /usr/include/c++/4.8.5/ios:43:0,
#7 14.39                  from /usr/include/c++/4.8.5/istream:38,
#7 14.39                  from /usr/include/c++/4.8.5/sstream:38,
#7 14.39                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/streambuf:802:7: error: 'std::basic_streambuf<_CharT, _Traits>::basic_streambuf(const std::basic_streambuf<_CharT, _Traits>&) [with _CharT = char; _Traits = std::char_traits<char>]' is private
#7 14.39        basic_streambuf(const basic_streambuf& __sb)
#7 14.39        ^
#7 14.39 In file included from /assimp/include/assimp/TinyFormatter.h:56:0,
#7 14.39                  from /assimp/include/assimp/Logger.hpp:50,
#7 14.39                  from /assimp/include/assimp/DefaultLogger.hpp:49,
#7 14.39                  from /assimp/code/Common/FileSystemFilter.h:50,
#7 14.39                  from /assimp/code/Common/BaseImporter.cpp:46:
#7 14.39 /usr/include/c++/4.8.5/sstream:64:11: error: within this context
#7 14.39      class basic_stringbuf : public basic_streambuf<_CharT, _Traits>
#7 14.39            ^
#7 14.52 make[2]: *** [code/CMakeFiles/assimp.dir/Common/BaseImporter.cpp.o] Error 1
#7 14.52 make[2]: *** Waiting for unfinished jobs....
#7 15.90 In file included from /assimp/include/assimp/Exceptional.h:50:0,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /assimp/include/assimp/TinyFormatter.h: In instantiation of 'Assimp::Formatter::basic_formatter<T, CharTraits, Allocator>::basic_formatter(Assimp::Formatter::basic_formatter<T, CharTraits, Allocator>&&) [with T = char; CharTraits = std::char_traits<char>; Allocator = std::allocator<char>]':
#7 15.90 /assimp/include/assimp/Exceptional.h:77:82:   required from 'DeadlyImportError::DeadlyImportError(T&& ...) [with T = {std::basic_string<char, std::char_traits<char>, std::allocator<char> >}]'
#7 15.90 /assimp/code/AssetLib/X3D/X3DImporter.hpp:64:88:   required from here
#7 15.90 /assimp/include/assimp/TinyFormatter.h:92:49: error: use of deleted function 'std::basic_ostringstream<char>::basic_ostringstream(const std::basic_ostringstream<char>&)'
#7 15.90          : underlying(std::move(other.underlying)) {
#7 15.90                                                  ^
#7 15.90 In file included from /assimp/include/assimp/TinyFormatter.h:56:0,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/sstream:387:11: note: 'std::basic_ostringstream<char>::basic_ostringstream(const std::basic_ostringstream<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 15.90      class basic_ostringstream : public basic_ostream<_CharT, _Traits>
#7 15.90            ^
#7 15.90 /usr/include/c++/4.8.5/sstream:387:11: error: use of deleted function 'std::basic_ostream<char>::basic_ostream(const std::basic_ostream<char>&)'
#7 15.90 In file included from /usr/include/c++/4.8.5/istream:39:0,
#7 15.90                  from /usr/include/c++/4.8.5/sstream:38,
#7 15.90                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/ostream:58:11: note: 'std::basic_ostream<char>::basic_ostream(const std::basic_ostream<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 15.90      class basic_ostream : virtual public basic_ios<_CharT, _Traits>
#7 15.90            ^
#7 15.90 /usr/include/c++/4.8.5/ostream:58:11: error: use of deleted function 'std::basic_ios<char>::basic_ios(const std::basic_ios<char>&)'
#7 15.90 In file included from /usr/include/c++/4.8.5/ios:44:0,
#7 15.90                  from /usr/include/c++/4.8.5/istream:38,
#7 15.90                  from /usr/include/c++/4.8.5/sstream:38,
#7 15.90                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/bits/basic_ios.h:66:11: note: 'std::basic_ios<char>::basic_ios(const std::basic_ios<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 15.90      class basic_ios : public ios_base
#7 15.90            ^
#7 15.90 In file included from /usr/include/c++/4.8.5/ios:42:0,
#7 15.90                  from /usr/include/c++/4.8.5/istream:38,
#7 15.90                  from /usr/include/c++/4.8.5/sstream:38,
#7 15.90                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/bits/ios_base.h:786:5: error: 'std::ios_base::ios_base(const std::ios_base&)' is private
#7 15.90      ios_base(const ios_base&);
#7 15.90      ^
#7 15.90 In file included from /usr/include/c++/4.8.5/ios:44:0,
#7 15.90                  from /usr/include/c++/4.8.5/istream:38,
#7 15.90                  from /usr/include/c++/4.8.5/sstream:38,
#7 15.90                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/bits/basic_ios.h:66:11: error: within this context
#7 15.90      class basic_ios : public ios_base
#7 15.90            ^
#7 15.90 In file included from /assimp/include/assimp/TinyFormatter.h:56:0,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/sstream:387:11: error: use of deleted function 'std::basic_ios<char>::basic_ios(const std::basic_ios<char>&)'
#7 15.90      class basic_ostringstream : public basic_ostream<_CharT, _Traits>
#7 15.90            ^
#7 15.90 /usr/include/c++/4.8.5/sstream:387:11: error: use of deleted function 'std::basic_stringbuf<char>::basic_stringbuf(const std::basic_stringbuf<char>&)'
#7 15.90 /usr/include/c++/4.8.5/sstream:64:11: note: 'std::basic_stringbuf<char>::basic_stringbuf(const std::basic_stringbuf<char>&)' is implicitly deleted because the default definition would be ill-formed:
#7 15.90      class basic_stringbuf : public basic_streambuf<_CharT, _Traits>
#7 15.90            ^
#7 15.90 In file included from /usr/include/c++/4.8.5/ios:43:0,
#7 15.90                  from /usr/include/c++/4.8.5/istream:38,
#7 15.90                  from /usr/include/c++/4.8.5/sstream:38,
#7 15.90                  from /assimp/include/assimp/TinyFormatter.h:56,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/streambuf:802:7: error: 'std::basic_streambuf<_CharT, _Traits>::basic_streambuf(const std::basic_streambuf<_CharT, _Traits>&) [with _CharT = char; _Traits = std::char_traits<char>]' is private
#7 15.90        basic_streambuf(const basic_streambuf& __sb)
#7 15.90        ^
#7 15.90 In file included from /assimp/include/assimp/TinyFormatter.h:56:0,
#7 15.90                  from /assimp/include/assimp/Exceptional.h:50,
#7 15.90                  from /assimp/include/assimp/BaseImporter.h:52,
#7 15.90                  from /assimp/code/Common/ImporterRegistry.cpp:49:
#7 15.90 /usr/include/c++/4.8.5/sstream:64:11: error: within this context
#7 15.90      class basic_stringbuf : public basic_streambuf<_CharT, _Traits>
#7 15.90            ^
#7 16.14 make[2]: *** [code/CMakeFiles/assimp.dir/Common/ImporterRegistry.cpp.o] Error 1
#7 16.14 make[1]: *** [code/CMakeFiles/assimp.dir/all] Error 2
#7 16.14 make: *** [all] Error 2
#7 ERROR: executor failed running [/bin/sh -c git clone https://github.com/assimp/assimp.git &&         cd assimp &&    git reset --hard $ASSIMP_HASH &&       cmake CMakeLists.txt -DBUILD_SHARED_LIBS=OFF -DASSIMP_BUILD_TESTS=OFF &&        make -j4]: exit code: 2
------
 > [4/4] RUN git clone https://github.com/assimp/assimp.git &&  cd assimp &&   git reset --hard 0c9947f8b9012faa43a77917f1035db2ffc9a0d2 &&     cmake CMakeLists.txt -DBUILD_SHARED_LIBS=OFF -DASSIMP_BUILD_TESTS=OFF &&        make -j4:
------
executor failed running [/bin/sh -c git clone https://github.com/assimp/assimp.git &&   cd assimp &&    git reset --hard $ASSIMP_HASH &&        cmake CMakeLists.txt -DBUILD_SHARED_LIBS=OFF -DASSIMP_BUILD_TESTS=OFF &&        make -j4]: exit code: 2
</pre>

**Expected behavior**
No errors should have occurred, and we should be able to see a binary saved at `/assimp/bin`, which could be called from the terminal. This error is not present with an earlier version of Assimp: `ARG ASSIMP_HASH="a2186a05536a6ba0def03c72c569ab581fc1e573"`

**Desktop that ran Docker:**
 - OS: Windows 10 64-Bit

**Additional context**
I can make this to work successfully when I use an old assimp commit `a2186a05536a6ba0def03c72c569ab581fc1e573`. However, with the latest master, I keep getting the errors you see above.

**Comment by turol** [comment]
You're trying to use an ancient g++ (4.8.5). I don't think this is supported any more. Also you gcc (7.2.1) and g++ (4.8.5) don't match. This might also cause trouble. Update your GCC packages.

**Comment by raphaelsetin** [comment]
@turol you were totally right! I didn't realize that I had to update these packages nor that they were the reason of the build failure.

After a little bit more research, I found that adding the package `gcc72-c++.x86_64` -- that's the latest at the moment -- at the first `yum install` command of the Dockerfile did the trick for me. Now I can successfully build master 🎉 

In case future people bump into this problem and want to know what's the latest `gcc` package available for their system, then execute the following `yum list | grep gcc` and grab the most up to date version.
